Gems/AtomLyIntegration/AtomViewportDisplayIcons/Code/Source/AtomViewportDisplayIconsSystemComponent.cpp

```diff
@@ -172,13 +172,16 @@ namespace AZ::Render
         }
         else if (drawParameters.m_positionSpace == CoordinateSpace::WorldSpace)
         {
+            // Calculate the ndc point (0.0-1.0 range) including depth
+            const AZ::Vector3 ndcPoint = AzFramework::WorldToScreenNdc(
+                drawParameters.m_position, viewportContext->GetCameraViewMatrixAsMatrix3x4(),
+                viewportContext->GetCameraProjectionMatrix());
+
             // Calculate our screen space position using the viewport size
             // We want this instead of RenderViewportWidget::WorldToScreen which works in QWidget virtual coordinate space
-            const AzFramework::ScreenPoint position = AzFramework::WorldToScreen(
-                drawParameters.m_position, viewportContext->GetCameraViewMatrixAsMatrix3x4(),
-                viewportContext->GetCameraProjectionMatrix(), viewportSize);
-            screenPosition.SetX(aznumeric_cast<float>(position.m_x));
-            screenPosition.SetY(aznumeric_cast<float>(position.m_y));
+            const AzFramework::ScreenPoint screenPoint = AzFramework::ScreenPointFromNdc(AZ::Vector3ToVector2(ndcPoint), viewportSize);
+
+            screenPosition = AZ::Vector2ToVector3(AzFramework::Vector2FromScreenPoint(screenPoint), ndcPoint.GetZ());
```

Comment:
nit:This seems ugly to go from a vec3 -> vec2 -> ScreenPoint -> vec3 (ScreenPoint + original Z)

Reply:
The shader could be updated to work strictly in clip space pretty easily, if desired (just don't do a transform on the input vector3).

Reply:
+1,  what about overloading WorldToScreen() and providing a function that returns a Vector3 so you get the depth?  

Also, I'm not really a fan of having these multiple types: WindowSize, ScreenSize, ScreenVector, ScreenPoint, all having their own m_width, m_height, m_x, m_y and specialized functions - e.g. ScreenSizeFromVector2 instead of a constructor that takes in a Vector2 const ref.  It makes for a lot of noisy code to convert between all those types when they all share a common underlying members (either uint x,y or float x,y).  Feels like over-engineering.  Also, the conversion from vector2 to uint32 involve floor ops etc.  are we really running into floating point issues for converting numbers in screen sizes or some SIMD math issues?

Reply:
This is outside the scope of the PR, I can tidy-up some of this code but it is very useful to have `ScreenPoint` and `ScreenVector` etc that are signed integral types (not floating point types). We always want these points to be in 'pixels' or 'points' and reusing `Vector3` is wrong for this reason. They also signal intent at the API boundary. This is code the viewport team has to own and maintain and I do not want to remove these types anytime soon. There are numerous advantages to having them around. As an aside, if we had a templated vector math type (`Vector<T, size>` we could reuse a lot of the useful functionality and not have to create our own types). Also a lot of these conversions are trivial and will be inlined away in release (I can do some tests in compiler explorer to verify later).

Reply:
I've added `Vector3FromScreenPoint` for fewer conversions but will leave the other conversions for now. I much prefer being explicit about the different types and the conversions are clear and easy to understand in the code rather than weird implicit conversions.

Reply:
> +1, what about overloading WorldToScreenNdc() with a function that returns a Vector3 so you get the depth?